samcli/lib/config/samconfig.py

```diff
@@ -41,6 +41,12 @@ def __init__(self, config_dir, filename=None):
         """
         self.filepath = Path(config_dir, filename or DEFAULT_CONFIG_FILE_NAME)
 
+    def get_env_names(self):
+        self._read()
+        if isinstance(self.document, dict):
+            return [env for env in self.document.keys() if env != VERSION_KEY]
```

Comment:
Shouldn't it instead check that `env` is a dictionary (or whatever represents a TOML table)? Otherwise this will break if there's another top-level key.

Reply:
TOML is like YAML; A hierarchy-based document. It is SAM who defined its own hierarchy to be: `[env / command_names / section / key]`. i.e. all the first-level keys in the toml file represents names of envs. (with one exception, the version).

Reply:
If tomorrow `samconfig.toml` gets a new top-level value `foo = "bar"`, this code breaks as it will include `foo` in the return values, no?

Reply:
If [SamConfig.put](https://github.com/aws/aws-sam-cli/blob/e4853b39bf8011132d08a12fdbb8c2e18437b9d9/samcli/lib/config/samconfig.py#L90) method changed they [way it persists the configuration](https://github.com/aws/aws-sam-cli/blob/e4853b39bf8011132d08a12fdbb8c2e18437b9d9/samcli/lib/config/samconfig.py#L120-L131), this method should change accordingly.

Reply:
Right, so `env` is a top-level dictionary. Something like this should be robust to changes:

```python
[k for k, v in self.document.items() if isinstance(v, dict)]
```

Reply:
Ah, you are talking about how to exclude the `Version` key. I thought you are questioning the whole approach.